lib/src/rules/type_test_on_generic_type_variable.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+
+import '../analyzer.dart';
+
+const _desc = r' ';
+
+const _details = r'''
+**DON'T** perform type tests on generic type variables.
+
+A type variable for a generic class or method will always be a `Type`. A type
```

Comment:
Copying @lrhn comment from the commit so it's on the PR instead.

https://github.com/dart-lang/linter/commit/b9d99dfddedbe051ac7f9d7eba0ad00b582a0e39#r119914277

Hmm.  A type variable isnt a `Type`, but it evaluates to one when used as an *expression*, not when used as a type.

Also "of that type" does not have another occurrence of "type" to refer back to. It refers to the type bound to the type variable, but a type variable is not a type (it's a name which can be used to refer to a type).

Maybe:

> A type variable, the type parameter of a generic class or method,
> used as an expression always evalautes to a `Type` object.
> The first operand of `is` or `as` type tests is an expression, so
> a type test of a type variable does not check any aspect
> of the type bound to that type variable. 
> It just tests an object of type `Type`.

Reply:
I took some of this wording and changed the last half. WDYT?